### PR TITLE
Improve wheels and sdist

### PR DESCRIPTION
**Pull request by T-256**
Replaces #192 

Excludes non-required and build-only files from bdist wheels.
Also, adds `tests` folder to sdist.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -73,11 +108,15 @@ def run(self):
         super().run()
 
 
+# this option is only valid in setup.py
+kwargs = {"cffi_modules": ["curl_cffi/build.py:ffibuilder"]}
+if len(sys.argv) > 1 and sys.argv[1] != 'bdist_wheel':
```
Comment (T-256) [inline setup.py:113]: I didn't find other way else to detect sdist-only calls

File: curl_cffi/build.py
Comment (T-256) [inline curl_cffi/build.py:1]: Per https://github.com/yifeikong/curl_cffi/pull/192#discussion_r1464481207
If you reviewed and approved changes applied to this file, then I'll move it to `scripts` folder before merge.

self-note: dependent [line](https://github.com/yifeikong/curl_cffi/pull/236/files#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7R112)

File: curl_cffi/build.py [hunk 2]
```diff
@@ -1,47 +1,76 @@
...
+elif system == "Darwin":
+    if machine in ("x86_64", "aarch64"):
+        libdir = "/Users/runner/work/_temp/install/lib"
+    else:
+        libdir = "ERROR"
 else:
-    libdir = os.path.expanduser("~/.local/lib")
+    if machine in ("x86_64", "arm", "aarch64"):
+        libdir = os.path.expanduser("~/.local/lib")
+    else:
```
Comment (yifeikong) [inline curl_cffi/build.py:47]: no need to detect here, there are no other archs for macOS

File: curl_cffi/build.py [hunk 3]
```diff
@@ -1,47 +1,76 @@
...
+    if machine == "x86_64":
         libdir = "./lib32"
-elif uname.system == "Darwin":
-    libdir = "/Users/runner/work/_temp/install/lib"
+    elif machine == "i686":
+        libdir = "./lib64"
+    else:
+        libdir = "ERROR"
+elif system == "Darwin":
+    if machine in ("x86_64", "aarch64"):
```
Comment (yifeikong) [inline curl_cffi/build.py:40]: no need to detect here, there are no other archs for macOS

File: setup.py [hunk 4]
```diff
@@ -73,11 +108,15 @@ def run(self):
         super().run()
 
 
+# this option is only valid in setup.py
```
Comment (yifeikong) [inline setup.py:111]: Why do we need to exclude this argument?

File: curl_cffi/build.py (hunk 3 above)
Comment (T-256) [reply curl_cffi/build.py:40]: return type of `abs_machine` is `Literal['x86_64', 'i686', 'aarch64', 'arm']`. let's be explicit on handling what it returns.

File: setup.py (hunk 4 above)
Comment (T-256) [reply setup.py:111]: I think it was in previous for making ci test passed. (because of sdist problem was existing on that time.)
Lemme try to revert it.

**Comment by yifeikong** [comment]
Are you on Windows? could please make sure that files are correctly marked as moved, not deleted and readded? It's very hard to track the diff.

**Comment by T-256** [comment]
please review commits one by one. latest commit in gh ui:
image

**Comment by yifeikong** [comment]
I just moved the files in the main branch, so that you can focus on modifying current files. I'm not sure why GitHub is not displaying moved files correctly, it's really weird.

File: setup.py [hunk 5]
```diff
@@ -23,85 +24,44 @@ def get_tag(self):
...
-
-    # x86 based archs
-    if machine in ('AMD64', 'x86_64', 'i686'):
-        return "x86_64" if is_64 else "i686"
-    # arm based archs
-    elif machine in ('aarch64', 'arm64', 'armv6l', 'armv7l'):
-        return "aarch64" if is_64 else "arm"
-    else:
-        raise Exception("Unsupported processor")
+def detect_arch():
```
Comment (T-256) [inline setup.py:27]: again we have duplicated detection !?

Comment (yifeikong) [reply setup.py]: Sorry for the misleading before, it turns out that we can download_so in build.py :)

**Pull request merged**